C#/MarsRoverKata/MarsRoverKata/Src/Grid.cs

```diff
@@ -0,0 +1,40 @@
+﻿namespace MarsRoverKata.Src
+{
+    class Grid
+    {
+        private readonly int _maxWidth;
+        private readonly int _maxHeight;
+
+        public Grid(int maxWidth, int maxHeight)
+        {
+            this._maxWidth = maxWidth;
+            this._maxHeight = maxHeight;
+        }
+
+        public GridPosition MoveNorth(GridPosition position) =>
+            WrapAroundIfNecessary(position.X, position.Y + 1);
+
+        public GridPosition MoveEast(GridPosition position) =>
+            WrapAroundIfNecessary(position.X + 1, position.Y);
+
+        public GridPosition MoveSouth(GridPosition position) =>
+            WrapAroundIfNecessary(position.X, position.Y - 1);
+
+        public GridPosition MoveWest(GridPosition position) =>
+            WrapAroundIfNecessary(position.X - 1, position.Y);
+
+        public GridPosition WrapAroundIfNecessary(int x, int y)
```

Comment:
Couldn't think of a good name for the boundary-checking-which-returns-an-altered-position-if-it-is-necessary method